src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (int i = 0; i < figures.length; i++) {
+            if (i < figures.length / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
+            figures[i].draw();
```

Comment:
just crate figure per loop and draw it 
```suggestion
        for (int i = 0; i < figures.length; i++) {
        Figure figer;
            if (i < figures.length / 2) {
                figure = figureSupplier.getRandomFigure();
            } else {
                figure = figureSupplier.getDefaultFigure();
            }
            figure.draw();
```